### ISSUE TITLE
Search: Focus loss when loading more search results in search modal.

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/jetpack#23397]
### Impacted plugin

Jetpack

### Steps to Reproduce

When the 'Load more' button is clicked, focus is lost, landing outside of the page content in the browser chrome.

### Possible Fix
Accessible options would include:

- "Load more" should retain focus, and give audible notice that new results have been loaded above, so the user knows they have to navigate in reverse to find the new values.

- "Load more" should retain focus, and new results should be loaded below the 'load more' button. Audible notice should be given that new results are loaded.

- "Load more" should retain focus, and insert a link following the button that will link the user to the point in the document where new results have been loaded.

### Other Information
This issue was surfaced to team 51 by an a11y contractor working on a partner site. 

### Browser

Other / Not applicable
